shockbot.py

```diff
@@ -59,73 +57,99 @@
 next_shock = datetime.now()
 next_vibe = datetime.now()
 
+# Define a general usage string
+USAGE = f'''Usage: /shock|vibrate <strength> <duration>
+Min, max shock strength: {SHOCK_STR_MIN} %, {SHOCK_STR_MAX} % 
+Min, max shock length: {SHOCK_DUR_MIN / 1000} s, {SHOCK_DUR_MAX / 1000} s
+Min, max vibe strength: {VIBE_STR_MIN} %, {VIBE_STR_MAX} %
+Min, max vibe length: {VIBE_DUR_MIN / 1000} s, {VIBE_DUR_MAX / 1000} s'''
+
+# This function handles the two actions
+async def command(event, cmd):
+    global next_shock
+    global next_vibe
+
+    # Get action and its parameters from the command args
+    action = cmd[0].split('/')[1].capitalize()
+    action_strength = cmd[1]
+    action_length = int(cmd[2] * 1000)
+
+    # Determine the correct length and strength depending on the type of the action
+    match action:
+        case 'Shock':
+            if action_strength > SHOCK_STR_MAX: action_strength = SHOCK_STR_MAX
+            if action_strength < SHOCK_STR_MIN: action_strength = SHOCK_STR_MIN
+            if action_length > SHOCK_DUR_MAX: action_length = SHOCK_DUR_MAX
+            if action_length < SHOCK_DUR_MIN: action_length = SHOCK_DUR_MIN
+        case 'Vibrate':
+            if action_strength > VIBE_STR_MAX: action_strength = VIBE_STR_MAX
+            if action_strength < VIBE_STR_MIN: action_strength = VIBE_STR_MIN
+            if action_length > VIBE_DUR_MAX: action_length = VIBE_DUR_MAX
+            if action_length < VIBE_DUR_MIN: action_length = VIBE_DUR_MIN
```

Comment:
This can be shortened with `min()` and `max()` functions instead of if statements.
```
case 'Shock':
    max(SHOCK_STR_MIN, min(SHOCK_STR_MAX, action_strength))
    max(SHOCK_DUR_MIN, min(SHOCK_DUR_MAX, action_length))
case 'Vibrate':
    max(VIBE_STR_MIN, min(VIBE_STR_MAX, action_strength))
    max(VIBE_DUR_MIN, min(VIBE_DUR_MAX, action_length))
```

Reply:
I actually hadn't touched python in ages :D But that would make it way more readable.

Reply:
Now that I think about it, I don't think these statements could be considered equivalent. I need to check whether the value falls in the interval of (VIBE_STR_MIN, VIBE_STR_MAX) and I cannot really see how I could accomplish this with the min()/max() functions.